parser_library/src/expressions/conditional_assembly/ca_expr_policy.cpp

```diff
@@ -312,8 +298,54 @@ ca_expr_ops get_expr_operator(const std::string& op)
 }
 
 ca_expr_ops ca_arithmetic_policy::get_operator(const std::string& symbol) { return get_expr_operator(symbol); }
+std::variant<std::monostate, invalid_by_policy, ca_expr_op> ca_arithmetic_policy::get_operator_properties(
+    const std::string& symbol)
+{
+    auto o = get_operator(symbol);
+    if (o == ca_expr_ops::UNKNOWN)
+        return std::monostate();
+    bool unary = is_unary(o);
+    bool binary = is_binary(o);
+    if (!unary && !binary)
+        return invalid_by_policy();
```

Comment:
Do you want to create a static function in `ca_common_expr_policy` that will do this set of checks as this part is duplicated 3 times?

Reply:
the is_unary and is_binary are specialized by the individual policies, so I cannot really do it that way, but I am considering just droping all those functions in favor of one switch(-like) code...

Reply:
Ok, I see. Other option could be having a base class (e.g. `ca_arithmetic_policy_base`) from which the other policies could be derived.